decidim-core/app/cells/decidim/activities_cell.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Decidim
+  # Renders a collection of activities using a different cell for
+  # each one.
+  class ActivitiesCell < Decidim::ViewModel
+    include Decidim::CardHelper
+    include Decidim::IconHelper
+    include Decidim::Core::Engine.routes.url_helpers
+
+    delegate :current_organization, to: :controller
+
+    # Since we're rendering each activity separatedly we need to trigger
+    # BatchLoader in order to accumulate all the ids to be found later.
+    def show
+      return if activities.blank?
+
+      activities.map do |activity|
+        activity.organization_lazy
+        activity.resource_lazy
+        activity.participatory_space_lazy
+        activity.component_lazy
+      end
+
+      render
+    end
+
+    def cell_for(activity)
+      "#{activity.resource_type.constantize.name.underscore}_activity"
```

Comment:
This code assumes the cell exists, what happens if the cell doesn't exist? Maybe you can do something like:

1. Build the constant name (`"#{activity.resource_type.constantize.name.underscore}_activity_cell".camelize.constantize)
1. Check if the constant exists
  1. If it does, return the cell name
  1. If it doesn't, return a default activity cell

Reply:
Alternatively:

1. Rename this method to `activity_cell_for(activity)` and make it so:

```
def activity_cell_for(activity)
  cell "#{activity.resource_type.constantize.name.underscore}_activity"), activity
rescue
  cell "decidim/activity", activity
end
```

Reply:
I'm not sure we can have a default activity cell, since it won't know how to render a given resource (it doesn't know if it has a title or where to link it to).

Reply:
Also, at `Decidim::ActivitySearch` there's a list of classes that will be searched as an activity, so it's unlikely that other resources are rendered.

Reply:
Ok!